cpp/ql/src/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -57,16 +57,20 @@ class Iterator extends Type {
   }
 }
 
+/**
+ * We use this instead of `getExplicitlyConverted` to ensure that the context can be pushed
+ * into `arg.getConversion*` and avoid a performance bottleneck.
+ */
```

Comment:
Do you think it's feasible to speed up `Expr.getExplicitlyConverted` itself so other queries don't run into this same problem? I think we could avoid the transitive closures and instead iterate from the unconverted `Expr` along each explicit conversion until we reach the last `Expr` on that chain.

Reply:
Yes. It's not as fast as with manual context-pushing, but I think it's much better than the current version (unless there's other predicates in the new recursion that I'm not noticing). I'll start a CPP-Differences job.

Reply:
https://github.com/github/codeql/pull/4213 has been merged and addresses this point.